src/cljs/main/broadfcui/main.cljs

```diff
@@ -204,6 +204,15 @@
   {:handle-hash-change
    (fn [{:keys [state]}]
      (let [window-hash (aget js/window "location" "hash")]
+       ;; if Terra redirects are globally enabled via config;
+       ;;   and the current hash has a path handler;
+       ;;   and that path handler has a Terra redirect,
+       ;; then redirect to Terra.
+       (when (config/terra-redirects-enabled)
+         (when-let [handler (nav/find-path-handler window-hash)]
+           (let [{:keys [make-props terra-redirect]} handler]
+             (when terra-redirect
+               (js-invoke (aget js/window "location") "replace" (str (config/terra-url) "/?fcredir=1#" (terra-redirect (make-props))))))))
```

Comment:
Could we also potentially use `fcredir` as a switch to show a "Welcome to FireCloud powered by Terra" message in Terra?

Reply:
sure! sounds good. That would also be good place in Terra UI to *strip* the `?fcredir=1` from the url so it isn't persistent across "pages".

Reply:
Agreed!